src/rules/sortImportsRule.ts

```diff
@@ -0,0 +1,244 @@
+            read the code and find necessary imports later. This rule is purely a matter of style.
+            
+            This rule checks all import declarations and verifies that all imports are first sorted by the used member 
+            syntax and then alphabetically by the first member or alias name.
+            `,
+    optionsDescription: Lint.Utils.dedent`
+      - \`"ignore-case"\` does case-insensitive comparisons (default: \`false\`)
+      - \`"ignore-member-sort"\` allows members in multiple type imports to occur in any order (default: \`false\`)
+      - \`"member-syntax-sort-order"\` (default: \`["none", "all", "multiple", "single", "alias"]\`); all 5 items must be 
+      present in the array, but you can change the order: 
+        - \`none\` = import module without exported bindings.
+        - \`all\` = import all members provided by exported bindings.
+        - \`multiple\` = import multiple members.
+        - \`single\` = import a single member.
+        - \`alias\` = creates an alias for a member. This is unique to TER and not in ESLint's \`sort-imports\`.
+      `,
+    options: {
+      type: 'object',
+      properties: {
+        'member-syntax-sort-order': {
+          type: 'array',
+          items: {
+            type: 'string',
+            enum: ['none', 'all', 'multiple', 'single', 'alias']
+          },
+          minLength: 5,
+          maxLength: 5
+        },
+        'ignore-case': {
+          type: 'boolean'
+        },
+        'ignore-member-sort': {
+          type: 'boolean'
+        }
+      }
+    },
+    optionExamples: [
+      Lint.Utils.dedent`
+        "${RULE_NAME}": [true]
+        `,
+      Lint.Utils.dedent`
+        "${RULE_NAME}": [true, { "ignore-case" }]
+        `,
+      Lint.Utils.dedent`
+        "${RULE_NAME}": [true, { "ignore-member-sort" }]
+        `,
+      Lint.Utils.dedent`
+        "${RULE_NAME}": [true, { "member-syntax-sort-order": ["all", "single", "multiple", "none", "alias"] }]
+        `
+    ],
+    typescriptOnly: false,
+    type: 'style'
+  };
+
+  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+    const walker = new RuleWalker(sourceFile, this.getOptions());
+    return this.applyWithWalker(walker);
+  }
+}
+
+enum MemberSyntaxType {
+  None,
+  All,
+  Multiple,
+  Single,
+  Alias
+}
+
+interface ImportMetadata {
+  memberSyntaxType: MemberSyntaxType;
+  sortValue: string;
+}
+
+class RuleWalker extends Lint.RuleWalker {
+  private ignoreCase: boolean;
+  private ignoreMemberSort: boolean;
+  private expectedOrder: MemberSyntaxType[];
+
+  private currentImportIndex = 0;
+  private currentSortValue: { sortValue: string, originalValue: string };
+  private caseConverter: (s: string) => string;
+
+  constructor(sourceFile: ts.SourceFile, options: Lint.IOptions) {
+    super(sourceFile, options);
+
+    const optionSet = this.getOptions()[0] || {};
+
+    // this.ignoreCase = optionSet['ignore-case'];
```

Comment:
Removed the comment. I looked, and `this.options` is private, but the changes in TSLint are pretty exciting.